src/java/org/apache/cassandra/concurrent/AbstractLocalAwareExecutorService.java

```diff
@@ -147,10 +145,12 @@ class FutureTask<T> extends SimpleCondition implements Future<T>, Runnable
         private boolean failure;
         private Object result = this;
         private final Callable<T> callable;
+        private final DetachedSpan detachedSpan;
 
         public FutureTask(Callable<T> callable)
         {
             this.callable = callable;
+            this.detachedSpan = DetachedSpan.start("FutureTask");
```

Comment:
@jackwickham does this span not need to be completed at some point?

Reply:
ah you might be right, this is an older version of the lib so I was trying to piece together the behaviour that we need, but actually digging further it looks like it used to be done differently